std/range/package.d

```diff
@@ -6904,21 +6904,27 @@ Indexed!(Source, Indices) indexed(Source, Indices)(Source source, Indices indice
 
 /**
 This range iterates over fixed-sized chunks of size $(D chunkSize) of a
-$(D source) range. $(D Source) must be a forward range. $(D chunkSize) must be
+$(D source) range. $(D Source) must be an input range. $(D chunkSize) must be
 greater than zero.
 
 If $(D !isInfinite!Source) and $(D source.walkLength) is not evenly
 divisible by $(D chunkSize), the back element of this range will contain
 fewer than $(D chunkSize) elements.
 
+If `Source` is an input range but not a forward range, the resulting range and
+chunks will be single-pass only: iterating over `front` will shrink the chunk
+such that subsequent invocations of `front` will no longer return the full
+chunk, and calling `popFront` on the outer range will invalidate any lingering
+references to previous values of `front`.
+
 Params:
     source = Range from which the chunks will be selected
     chunkSize = Chunk size
 
 See_Also: $(LREF slide)
 
-Returns: Forward range of all chunks with propagated bidirectionality,
-         conditional random access and slicing.
+Returns: Input range of all chunks with propagated forwardness,
+    bidirectionality, conditional random access and slicing.
 */
 struct Chunks(Source)
 if (isForwardRange!Source)
```

Comment:
Originally, this is what I did but it made the diff look really ugly, that's why I went with splitting it into two overloads.  But since you asked for it, I'll go back to doing it that way.